heft-plugins/heft-typescript-plugin/src/TypeScriptBuilder.ts

```diff
@@ -350,12 +362,21 @@ export class TypeScriptBuilder {
   public async _runWatchAsync(tool: ITypeScriptTool): Promise<void> {
     const {
       ts,
+      system,
       measureSync: measureTsPerformance,
       pendingOperations,
       rawDiagnostics,
       pendingTranspilePromises
     } = tool;
 
+    // Remove this when watchFile and watchDirectory have been overridden
+    if (ts.sys.getCurrentDirectory() !== system.getCurrentDirectory()) {
+      // This happens because ts.sys.watchFile and ts.sys.watchDirectory wrap a memoized working directory.
+      throw new Error(
```

Comment:
So will this just cause Heft to crash in some cases?

Reply:
On further inspection, this mismatch shouldn't actually break anything, so I can remove this check.